newrelic/resource_newrelic_nrql_alert_condition.go

```diff
@@ -0,0 +1,271 @@
+						},
+					},
+				},
+				Required: true,
+				MinItems: 1,
+			},
+			"value_function": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "single_value",
+				ValidateFunc: validation.StringInSlice([]string{"single_value", "sum"}, false),
+			},
+		},
+	}
+}
+
+func buildNrqlAlertConditionStruct(d *schema.ResourceData) *newrelic.AlertNrqlCondition {
+	termSet := d.Get("term").([]interface{})
+	terms := make([]newrelic.AlertConditionTerm, len(termSet))
+
+	for i, termI := range termSet {
+		termM := termI.(map[string]interface{})
+
+		terms[i] = newrelic.AlertConditionTerm{
+			Duration:     termM["duration"].(int),
+			Operator:     termM["operator"].(string),
+			Priority:     termM["priority"].(string),
+			Threshold:    termM["threshold"].(float64),
+			TimeFunction: termM["time_function"].(string),
+		}
+	}
+
+	query := newrelic.AlertNrqlQuery{}
+
+	if nrqlQuery, ok := d.GetOk("nrql.0.query"); ok {
+		query.Query = nrqlQuery.(string)
+	}
+
+	if sinceValue, ok := d.GetOk("nrql.0.since_value"); ok {
+		query.SinceValue = sinceValue.(string)
+	}
+
+	condition := newrelic.AlertNrqlCondition{
+		Name:          d.Get("name").(string),
+		Enabled:       d.Get("enabled").(bool),
+		Terms:         terms,
+		PolicyID:      d.Get("policy_id").(int),
+		Nrql:          query,
+		ValueFunction: d.Get("value_function").(string),
+	}
+
+	if attr, ok := d.GetOk("runbook_url"); ok {
+		condition.RunbookURL = attr.(string)
+	}
+
+	return &condition
+}
+
+func readNrqlAlertConditionStruct(condition *newrelic.AlertNrqlCondition, d *schema.ResourceData) error {
+	ids, err := parseIDs(d.Id(), 2)
+	if err != nil {
+		return err
+	}
+
+	policyID := ids[0]
+
+	d.Set("policy_id", policyID)
+	d.Set("name", condition.Name)
+	d.Set("runbook_url", condition.RunbookURL)
+	d.Set("enabled", condition.Enabled)
+	d.Set("value_function", condition.ValueFunction)
+	d.Set("nrql.0.Query", condition.Nrql.Query)
+	d.Set("nrql.0.SinceValue", condition.Nrql.SinceValue)
+
+	var terms []map[string]interface{}
+
+	for _, src := range condition.Terms {
+		dst := map[string]interface{}{
+			"duration":      src.Duration,
+			"operator":      src.Operator,
+			"priority":      src.Priority,
+			"threshold":     src.Threshold,
+			"time_function": src.TimeFunction,
+		}
+		terms = append(terms, dst)
+	}
+
+	if err := d.Set("term", terms); err != nil {
+		return fmt.Errorf("[DEBUG] Error setting alert condition terms: %#v", err)
+	}
+
+	return nil
+}
+
+func resourceNewRelicNrqlAlertConditionCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*newrelic.Client)
+	condition := buildNrqlAlertConditionStruct(d)
+
+	log.Printf("[INFO] Creating New Relic NRQL alert condition %s", condition.Name)
+
+	condition, err := client.CreateAlertNrqlCondition(*condition)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(serializeIDs([]int{condition.PolicyID, condition.ID}))
+
+	return nil
```

Comment:
This has been modified to read the API per your recommendation. Acceptance tests re-ran successfully:
make testacc TEST=./newrelic TESTARGS='-run=TestAccNewRelicNrqlAlertCondition_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./newrelic -v -run=TestAccNewRelicNrqlAlertCondition_Basic -timeout 120m
=== RUN   TestAccNewRelicNrqlAlertCondition_Basic
--- PASS: TestAccNewRelicNrqlAlertCondition_Basic (7.47s)
PASS
ok  	github.com/terraform-providers/terraform-provider-newrelic/newrelic	8.080s